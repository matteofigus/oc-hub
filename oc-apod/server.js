const fetch = require('node-fetch');

export const data = (context, callback) => {
  const apiKey = context.params.apiKey;

  const url = `https://api.nasa.gov/planetary/apod?api_key=${apiKey}`;
  fetch(url)
    .then((response) => {
      if (response.status !== 200) {
        throw new Error(`url: ${response.url} status: ${response.status} statusText: ${response.statusText}`);
      }
      return response.json();
    })
    .then((json) => {
      callback(null, json);
    })
    .catch((error) => {
      callback(null, error);
    });
};
