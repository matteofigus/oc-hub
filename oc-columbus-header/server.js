export const data = (context, callback) => {
  const model = {
    logo: {
      href: '',
      url: context.params.logoUrl,
      alt: `${context.params.title} logo`,
    },
    title: context.params.title
  };
  callback(null, model);
};
