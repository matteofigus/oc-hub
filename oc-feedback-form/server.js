/* eslint-disable func-names, no-var */
module.exports.data = function (context, callback) {
  var model = { baseUrl: context.baseUrl, staticPath: context.staticPath, params: context.params };
  callback(null, model);
};
