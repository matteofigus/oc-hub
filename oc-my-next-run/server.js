const React = require('react');
const { renderToString } = require('react-dom/server');
const { ServerStyleSheet } = require('styled-components');

const App = require('./src/App').default;
const runs = require('./package.json').runs;

const makeWhen = (dateTime) => {
  if (typeof dateTime === 'object') {
    return new Date(...dateTime);
  }
  return new Date(dateTime);
};

export const data = (context, callback) => {
  const run = runs[context.params.id];
  const model = {
    twitterUsername: 'mattiaerre',
    when: makeWhen(run['date-time']),
    where: {
      googleMapsShortUrl: run['google-maps-short-url']
    },
    note: run.note
  };
  const props = { model };
  const app = renderToString(<App {...props} />);
  const sheet = new ServerStyleSheet();
  const css = sheet.getStyleTags();
  callback(null, {
    app,
    model,
    css,
    staticPath: context.staticPath
  });
};
