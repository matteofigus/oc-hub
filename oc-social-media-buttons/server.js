const React = require('react');
const { renderToString } = require('react-dom/server');
const { ServerStyleSheet } = require('styled-components');
const App = require('./src/App').default;

export const data = (context, callback) => {
  const app = renderToString(<App />);
  const sheet = new ServerStyleSheet();
  const css = sheet.getStyleTags();
  callback(null, { app, css });
};
