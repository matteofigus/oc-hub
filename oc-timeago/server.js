import moment from 'moment';

export const data = (context, callback) => { // eslint-disable-line import/prefer-default-export
  // https://momentjs.com/docs/#/displaying/fromnow/
  // https://momentjs.com/docs/#/displaying/from/
  const start = context.params.start;
  const end = context.params.end;

  const timeago = end ? moment(start).from(moment(end)) : moment(start).fromNow();

  return callback(null, { timeago });
};
