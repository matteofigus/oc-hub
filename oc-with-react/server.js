import React from 'react';
import { renderToString } from 'react-dom/server';
import App from './src/App';

module.exports.data = (context, callback) => {
  const app = renderToString(<App />);
  callback(null, {
    app,
    staticPath: context.staticPath
  });
};
