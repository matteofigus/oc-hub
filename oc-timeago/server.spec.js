import moment from 'moment';
import { data } from './server';

const now = 1487830758201; // 2017-02-22
Date.now = jest.fn(() => now);

const scenarios = [
  {
    start: '2016-02-22',
    end: '2017-02-22',
    timeago: 'a year ago'
  },
  {
    start: '2017-02-21',
    end: '2017-02-22',
    timeago: 'a day ago'
  },
  {
    start: '2017-02-20',
    end: '2017-02-22',
    timeago: '2 days ago'
  },
  {
    start: '2016-11-22',
    end: '2017-02-22',
    timeago: '3 months ago'
  },
  {
    start: '2015-08-17',
    timeago: '2 years ago'
  }
];

scenarios.forEach((scenario) => {
  test(`when? "${scenario.timeago}"`, () => {
    const context = { params: { start: scenario.start, end: scenario.end } };

    data(context, (error, model) => {
      expect(model.timeago).toBe(scenario.timeago);
    });
  });
});

test('w/ "end"', () => {
  const context = { params: { start: '2017-01-22' } };

  data(context, (error, model) => {
    expect(model).toMatchSnapshot();
  });
});

test('w/o "end"', () => {
  const context = { params: { start: '2017-01-22', end: moment() } };

  data(context, (error, model) => {
    expect(model).toMatchSnapshot();
  });
});
