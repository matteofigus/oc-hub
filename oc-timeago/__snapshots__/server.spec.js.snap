// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`w/ "end" 1`] = `
Object {
  "timeago": "a month ago",
}
`;

exports[`w/o "end" 1`] = `
Object {
  "timeago": "a month ago",
}
`;
